src/torchpme/lib/kspace_filter.py

```diff
@@ -188,6 +188,14 @@ def forward(self, mesh_values: torch.Tensor) -> torch.Tensor:
             s=mesh_values.shape[-3:],
         )
 
+        if torch.isnan(result).any():
+            raise ValueError(
+                "NaNs found in the output of the k-space filter. This is (likely) due "
+                "to a bad setting of `mesh_spacing`. Please try a different value."
```

Comment:
Maybe we give the user provided value in the error message?

```suggestion
                "The result of the k-space filter are NaN. This is (likely) due "
                f"to the `mesh_spacing` which is {mesh_spacing}."
```

Reply:
Would be good, but `mesh_spacing` isn't accessible in `KSpaceFilter`, only `ns_mesh` is available. Or we can move the check to there
https://github.com/lab-cosmo/torch-pme/blob/97fb4dc248298cce39977f246222aa8a92404341/src/torchpme/calculators/pme.py#L119-L121

Reply:
Arghhh, maybe just recompute them using the cell xD.